ci.yml

```diff
@@ -67,23 +67,33 @@ jobs:
                 params:
                     submodules: [build/terraform]
               - get: ci
-            # TODO(@ndmckinley): This task is too monolithic and should be split up.
-          - task: generate
-            file: ci/magic-modules/generate.yml
-            params:
-                GH_USERNAME: ((username))
+            # consumes: ci, magic-modules (detached HEAD)
+            # produces: magic-modules-branched (new branch, with submodule)
+          - task: branch-magic-modules
+            file: ci/magic-modules/branch.yml
+            # consumes: ci, magic-modules-branched
+            # produces: terraform-generated
+          - task: generate-terraform
+            file: ci/magic-modules/generate-terraform.yml
```

Comment:
I know right now you're focusing on getting TF working, but I'm curious on how this would look like eventually. Various steps are explicitly defined for TF. How would this expand when all Puppet/Chef submodules + Ansible are added here? Do you have an idea already? Or are you going to deal with that whenever you get there?

Reply:
You've nailed my next step.  :)  Once these are merged and all development moves to the magic-modules repo, I'll be adding those.  It'll be an `aggregate`, here - we'll generate all the repos in parallel.  Everywhere where it currently says "terraform", imagine a parallel "ansible", or "puppet/compute".  That's why, instead of generating a magic-modules folder, this now generates a terraform folder, which is sync'd below.  The sync below will be the fan-in, to the `aggregate`'s fan-out.